vault/core_cache_invalidate.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2025 OpenBao a Series of LF Projects, LLC
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/openbao/openbao/helper/namespace"
+)
+
+func (c *Core) Invalidate(key string) {
+	ctx, cancel := context.WithTimeout(c.activeContext, 2*time.Second)
+	defer cancel()
```

Comment:
I don't think the downstream code is checking for `ctx.Done()` ... so this may not have any real effect.

Reply:
@suprjinx In the current downstream code, no. This was to guard against plugins (to which invalidation events aren't yet routed) using invalidation to perform lots of actual storage operations (which does check context cancellation).